domain/keyupdater/service/service_test.go

```diff
@@ -102,3 +103,28 @@ func (s *serviceSuite) TestAuthorisedKeysForMachineNotFound(c *gc.C) {
 	)
 	c.Check(err, jc.ErrorIs, machineerrors.NotFound)
 }
+
+// TestGetInitialAuthorisedKeysForContainerSuccess tests the happy path for
+// Service.GetInitialAuthorisedKeysForContainer.
+func (s *serviceSuite) TestGetInitialAuthorisedKeysForContainerSuccess(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().AllAuthorisedKeys(gomock.Any()).Return(controllerKeys, nil)
+
+	keys, err := NewService(s.controllerKeyProvider, s.state).
+		GetInitialAuthorisedKeysForContainer(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(keys, jc.DeepEquals, controllerKeys)
+}
+
+// TestGetInitialAuthorisedKeysForContainerSuccess checks that
+// Service.GetInitialAuthorisedKeysForContainer surfaces errors from state.
+func (s *serviceSuite) TestGetInitialAuthorisedKeysForContainerFailure(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().AllAuthorisedKeys(gomock.Any()).Return(nil, errors.NotFound)
```

Comment:
I don't like returning NotFound here as it isn't a defined error for this func. Can we just call errors.New("foo")?